Content.Client/Overlays/EntityHealthBarOverlay.cs

```diff
@@ -52,31 +58,28 @@ protected override void Draw(in OverlayDrawArgs args)
             out var damageableComponent,
             out var spriteComponent))
         {
-            if (_entManager.TryGetComponent<MetaDataComponent>(uid, out var metaDataComponent) &&
-                metaDataComponent.Flags.HasFlag(MetaDataFlags.InContainer))
-            {
+            var metaDataComponent = _entManager.GetComponent<MetaDataComponent>(uid);
+            if (metaDataComponent.Flags.HasFlag(MetaDataFlags.InContainer))
+                continue;
+
+            // We want the stealth user to still be able to see his health bar himself
+            var viewer = _playerManager.LocalSession?.AttachedEntity;
+            if (viewer != uid && _entManager.TryGetComponent<StealthComponent>(uid, out var stealth) && stealth.Enabled)
```

Comment:
It might be worth just using `ExamineAttemptEvent`. I don't think there'd be a situation where you should be able to see their health if you can't also open the examine tooltip on them? and it already does stealth checks & exempts the cloaked entity.

Reply:
alright made it use the status icon system directly to indicate the checks should be the same

Reply:
>  seems a little wasteful to raise them instead of just doing the checks manually

Its probably fine performance wise? Unless you end up with a huge number of players on screen. Unless I'm significantly underestimating how expensive the InRangeUnoccluded check is